packages/arb-token-bridge-ui/src/hooks/useSigners.ts

```diff
@@ -0,0 +1,68 @@
+import { useMemo } from 'react'
+import { useWallet } from '@gimmixorg/use-wallet'
+import { providers } from 'ethers'
+import { JsonRpcSigner } from '@ethersproject/providers/lib/json-rpc-provider'
+
+import { useNetworks, UseNetworksStatus } from './useNetworks'
+import { rpcURLs } from '../util/networks'
+
+export enum UseSignersStatus {
+  NETWORK_NOT_CONNECTED = 'network_not_connected',
+  NETWORK_NOT_SUPPORTED = 'network_not_supported',
+  SUCCESS = 'success'
+}
+
+export type UseSignersDataUnknown = {
+  l1Signer: undefined
+  l2Signer: undefined
+}
+
+export type UseSignersData = {
+  l1Signer: JsonRpcSigner
+  l2Signer: JsonRpcSigner
+}
+
+export type UseSignersResult =
+  | ({ status: UseSignersStatus.NETWORK_NOT_CONNECTED } & UseSignersDataUnknown)
+  | ({ status: UseSignersStatus.NETWORK_NOT_SUPPORTED } & UseSignersDataUnknown)
+  | ({ status: UseSignersStatus.SUCCESS } & UseSignersData)
+
+const defaults: UseSignersDataUnknown = {
+  l1Signer: undefined,
+  l2Signer: undefined
+}
+
+export function useSigners(): UseSignersResult {
+  const networks = useNetworks()
+  const { account: address, provider } = useWallet()
+
+  return useMemo(() => {
+    if (networks.status === UseNetworksStatus.NOT_CONNECTED) {
+      return { status: UseSignersStatus.NETWORK_NOT_CONNECTED, ...defaults }
+    }
+
+    if (networks.status === UseNetworksStatus.NOT_SUPPORTED) {
+      return { status: UseSignersStatus.NETWORK_NOT_SUPPORTED, ...defaults }
+    }
+
+    const { l1Network, l2Network, isConnectedToArbitrum } = networks
+    const partnerNetwork = isConnectedToArbitrum ? l1Network : l2Network
+    const partnerProvider = new providers.JsonRpcProvider(
+      partnerNetwork.rpcURL || rpcURLs[partnerNetwork.chainID]
+    )
+
+    if (isConnectedToArbitrum) {
+      return {
+        status: UseSignersStatus.SUCCESS,
+        l1Signer: partnerProvider.getSigner(address!),
+        l2Signer: provider?.getSigner(0) as JsonRpcSigner
```

Comment:
Instead of `as JsonRpcSigner`, could we have the useSigners hook return null if no `provider` is returned? The null return value would only need to be handled in one place in the Injector (not convinced that my suggestion is best, just thinking about ways to avoid type coercion here)

Reply:
Added a check for `provider` being undefined [here](https://github.com/OffchainLabs/arb-token-bridge/pull/270/commits/077084b30da456ee10a5e299c1cd5d5eee140bba), but we'll need to keep `as JsonRpcSigner` until we get rid of `arb-ts` because of version clashes.